ibis/backends/clickhouse/tests/test_functions.py

```diff
@@ -532,3 +532,15 @@ def test_count_distinct_with_filter(alltypes):
     expected = alltypes.string_col.execute()
     expected = expected[expected.astype('int64') > 1].nunique()
     assert result == expected
+
+
+def test_group_concat(con, alltypes, translate):
+    t = alltypes
+    expr = t.string_col.group_concat(',')
+    expected = """arrayStringConcat(groupArray(`string_col`), ',')"""
+    assert translate(expr) == expected
+
+    expr = t.string_col.group_concat(sep='-', where=t.bool_col == 0)
+    expected = """arrayStringConcat(groupArray(\
+CASE WHEN `bool_col` = 0 THEN `string_col` ELSE Null END), '-')"""
```

Comment:
No big deal, but I think this approach is more readable (maybe just to me):

```suggestion
    expected = ("arrayStringConcat(groupArray("
                "CASE WHEN `bool_col` = 0 THEN `string_col` ELSE Null END), '-')")
```

Also, it'd probably be nice to have this as a separate test, so when it fails, we know which of the two asserts is failing faster.